internal/exec/go_getter_utils.go

```diff
@@ -56,77 +58,143 @@ func IsValidScheme(scheme string) bool {
+		}
 	}
 
+	l.Debug(fmt.Sprintf("url = %q:", src))
+
 	parsedURL, err := url.Parse(src)
 	if err != nil {
-		u.LogDebug(fmt.Sprintf("Failed to parse URL %q: %v\n", src, err))
+		l.Debug(fmt.Sprintf("Failed to parse URL %q: %v", src, err))
 		return "", false, fmt.Errorf("failed to parse URL %q: %w", src, err)
 	}
 
-	if strings.ToLower(parsedURL.Host) != "github.com" {
-		u.LogDebug(fmt.Sprintf("Host is %q, not 'github.com', skipping token injection\n", parsedURL.Host))
-		return "", false, nil
+	// If the URL uses the SSH scheme, check for an active SSH agent.
+	if parsedURL.Scheme == "ssh" && os.Getenv("SSH_AUTH_SOCK") == "" {
+		return "", false, fmt.Errorf("SSH URL detected but no SSH agent appears to be active. Please ensure your SSH key is loaded (e.g. run 'eval $(ssh-agent -s)' and 'ssh-add ~/.ssh/id_ed25519')")
 	}
 
-	parts := strings.SplitN(parsedURL.Path, "/", 4)
-	if len(parts) < 3 {
-		u.LogDebug(fmt.Sprintf("URL path %q doesn't look like /owner/repo\n", parsedURL.Path))
-		return "", false, fmt.Errorf("invalid GitHub URL %q", parsedURL.Path)
+	// Adjust host check to support GitHub, Bitbucket, GitLab, etc.
+	host := strings.ToLower(parsedURL.Host)
+	if host != "github.com" && host != "bitbucket.org" && host != "gitlab.com" {
+		l.Debug(fmt.Sprintf("Host is %q, not recognized for token injection", parsedURL.Host))
+		// For unrecognized hosts, simply return without injecting tokens.
+		return "", false, nil
 	}
 
-	atmosGitHubToken := os.Getenv("ATMOS_GITHUB_TOKEN")
-	gitHubToken := os.Getenv("GITHUB_TOKEN")
-
-	var usedToken string
-	var tokenSource string
-
-	// 1. If ATMOS_GITHUB_TOKEN is set, always use that
-	if atmosGitHubToken != "" {
-		usedToken = atmosGitHubToken
-		tokenSource = "ATMOS_GITHUB_TOKEN"
-		u.LogDebug("ATMOS_GITHUB_TOKEN is set\n")
-	} else {
-		// 2. Otherwise, only inject GITHUB_TOKEN if cfg.Settings.InjectGithubToken == true
-		if d.AtmosConfig.Settings.InjectGithubToken && gitHubToken != "" {
-			usedToken = gitHubToken
+	// TBC: should we support more tokens for Bitbucket and GitLab at all? Any other hosts?
+	// Any other git-enabled hosts to be added?
+	var token, tokenSource string
+	switch host {
+	case "github.com":
+		token = os.Getenv("ATMOS_GITHUB_TOKEN")
+		if token == "" && d.AtmosConfig.Settings.InjectGithubToken {
+			token = os.Getenv("GITHUB_TOKEN")
 			tokenSource = "GITHUB_TOKEN"
-			u.LogTrace("InjectGithubToken=true and GITHUB_TOKEN is set, using it\n")
 		} else {
-			u.LogTrace("No ATMOS_GITHUB_TOKEN or GITHUB_TOKEN found\n")
+			tokenSource = "ATMOS_GITHUB_TOKEN"
+		}
+	case "bitbucket.org":
+		token = os.Getenv("ATMOS_BITBUCKET_TOKEN")
+		if token == "" {
+			token = os.Getenv("BITBUCKET_TOKEN")
+			tokenSource = "BITBUCKET_TOKEN"
+		} else {
+			tokenSource = "ATMOS_BITBUCKET_TOKEN"
+		}
+	case "gitlab.com":
+		token = os.Getenv("ATMOS_GITLAB_TOKEN")
+		if token == "" {
+			token = os.Getenv("GITLAB_TOKEN")
+			tokenSource = "GITLAB_TOKEN"
```

Comment:
Reverse the order of token and token source, then os.Getenv should read tokenSouce